crates/cdk-sql-common/src/mint/mod.rs

```diff
@@ -219,6 +219,43 @@ where
             .execute(&self.inner)
             .await?;
 
+        if new_state == State::Spent {
+            // Update keyset_amounts table for redeemed amounts
+            // First, try UPDATE (most common case)
+            let updated = query(
+                r#"
+                UPDATE keyset_amounts
+                SET total_redeemed = total_redeemed + amounts.total
+                FROM (
+                    SELECT keyset_id, COALESCE(SUM(amount), 0) as total
+                    FROM proof
+                    WHERE y IN (:ys)
+                    GROUP BY keyset_id
+                ) amounts
+                WHERE keyset_amounts.keyset_id = amounts.keyset_id
```

Comment:
What if we update the `get_current_states` to return the `amount` and `keyset_id` with the y this way we can avoid the second query on the proofs table here.

Reply:
I think if we do this we need to make sure the get current state locks the rows affected to avoid a race condition.